### ISSUE TITLE
AWS docker: no video

### ISSUE THREAD

**Issue by grinono** [robertoandrade/docker-jitsi-meet#2]
Somehow when i'm installing this docker. Video/Audio is not working. I do see the other party online. Tried everything at the port level but does not seem to work. 

any idea how to fix this?

**Comment by robertoandrade** [comment]
You're running this on an EC2 machine? Try running the container with privileged networking (`-net host`) and make sure to forward all UDP traffic on ports 10000-20000 through the security group rules.

Another way I've got around that, in case you have other restrictions around UDP traffic on the client side, was to use the "preconfigured" tag of the image, in which I configure JVB to relay traffic in TCP mode over 443 alongside with the `jitsi-meet` web app. Use the following `sip-communicator.properties`:

```
org.jitsi.impl.neomedia.transform.srtp.SRTPCryptoContext.checkReplay=false
org.jitsi.videobridge.NAT_HARVESTER_LOCAL_ADDRESS=10.0.0.3
org.jitsi.videobridge.NAT_HARVESTER_PUBLIC_ADDRESS=153.92.34.62
org.jitsi.videobridge.TCP_HARVESTER_PORT=443
org.jitsi.videobridge.TCP_HARVESTER_MAPPED_PORT=443
org.jitsi.videobridge.rest.jetty.host=::
org.jitsi.videobridge.rest.jetty.port=443
org.jitsi.videobridge.rest.jetty.ProxyServlet.hostHeader=localhost
org.jitsi.videobridge.rest.jetty.ProxyServlet.pathSpec=/http-bind
org.jitsi.videobridge.rest.jetty.ProxyServlet.proxyTo=http://localhost:5280/http-bind
org.jitsi.videobridge.rest.jetty.ResourceHandler.resourceBase=/usr/share/jitsi-meet
org.jitsi.videobridge.rest.jetty.ResourceHandler.alias./config.js=/etc/jitsi/meet/localhost-config.js
org.jitsi.videobridge.rest.jetty.RewriteHandler.regex=^/([a-zA-Z0-9]+)$
org.jitsi.videobridge.rest.jetty.RewriteHandler.replacement=/
org.jitsi.videobridge.rest.jetty.tls.port=443
org.jitsi.videobridge.rest.jetty.sslContextFactory.keyStorePath=/etc/ssl/nginx/localhost.jks
org.jitsi.videobridge.rest.jetty.sslContextFactory.keyStorePassword=changeit
```

You can keep this on the host with the proper settings for local/public IP and mount the host directory where you have the file to `/etc/jitsi/videobridge`.

Same for `/etc/ssl/nginx`, make sure to have the `localhost.key` and `localhost.cer` on the host folder you're going to mount into the container and it should take care of generating the `.jks` with the default password, unless you want to generate your own and provide the associated password in the `.properties` file.

Here's an example container creation doing the mounting of the corresponding directories (this assumes you have 2 directories on the host respectively `.config` and `.ssl` containing both the `.properties` file and the `.key` and `.cer` as well as corresponding `.jks`):

```
export IMAGE=robertoandrade/jitsi-meet:preconfigured

docker create --name jitsi-meet \
    --net host \
    -v `pwd`/.config:/etc/jitsi/videobridge \
    -v `pwd`/.ssl:/etc/ssl/nginx \
    $IMAGE

docker start jitsi-meet
```

**Comment by grinono** [comment]
Thanks Roberto, i got it working on AWS now. But restrictive firewall that only allow 443 TCP is still an issue. (same for default config) Do you have any idea how to implement the two IP;s solution to get ICE and videobridge to relay via TCP 443?

**Comment by robertoandrade** [comment]
Yep, that's what I had mentioned in my previous comment. I have that in a separate branch right now but it's basically configuring JVB to serve meet over HTTPS (via jetty instead of nginx) as well as relay TURN traffic over TCP/443.

**Comment by atyenoria** [comment]
Try checking /etc/hosts in docker container.
If I change it properly, It works perfectly.

**Comment by lokeshh** [comment]
@robertoandrade How do I find the local IP address of my EC2?

> You can keep this on the host with the proper settings for local/public IP and mount the host directory where you have the file to /etc/jitsi/videobridge.

**Comment by sribi** [comment]
@atyenoria 
could you quickly point me into the direction of what should be specified in /etc/hosts?

thanks!
